### ISSUE TITLE
[Bug]: Errors Inserting into Distributed Hypertable: ERROR:  XX000: tuple already updated by self

### ISSUE THREAD

**Issue by wrona100** [timescale/timescaledb#5905]
### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Query executor

### What happened?

I get an error when trying to insert any line.
Table definition:
CREATE TABLE IF NOT EXISTS public.sec
(
    tagno integer NOT NULL,
    "timestamp" timestamp with time zone NOT NULL,
    value real,
    confidence smallint DEFAULT 100
)
query:
INSERT into public.sec(tagno, "timestamp", value, confidence)VALUES (73116, '2023-07-28 09:08:21+00', 1406.307739,100)
The problem is not with the configuration. The new identical table has no such problems.
I'm stuck, I don't want to create a new table. Restoring from a backup is not a solution - unless it was an isolated case.

The problem started to appear after I inserte data into table from two sources at the same time. But that may be a coincidence.

### TimescaleDB version affected

2.11

### PostgreSQL version used

14.7

### What operating system did you use?

Red Hat Enterprise Linux 8

### What installation method did you use?

Docker

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
2023-07-31 08:49:19.432 UTC [215903] DEBUG:  00000: StartTransaction(1) name: unnamed; blockState: DEFAULT; state: INPROGRESS, xid/subid/cid: 0/1/0
2023-07-31 08:49:19.432 UTC [215903] LOCATION:  ShowTransactionStateRec, xact.c:5436
2023-07-31 08:49:19.432 UTC [215903] STATEMENT:  INSERT into public.sec(tagno, "timestamp", value, confidence)VALUES (73116, '2023-07-28 09:08:21+00', 1406.307739,100);
2023-07-31 08:49:19.435 UTC [215903] ERROR:  XX000: tuple already updated by self
2023-07-31 08:49:19.435 UTC [215903] LOCATION:  simple_heap_update, heapam.c:4260
2023-07-31 08:49:19.435 UTC [215903] STATEMENT:  INSERT into public.sec(tagno, "timestamp", value, confidence)VALUES (73116, '2023-07-28 09:08:21+00', 1406.307739,100);
2023-07-31 08:49:19.435 UTC [215903] DEBUG:  00000: cleaned up 0 connections and 0 results at abort of transaction
2023-07-31 08:49:19.435 UTC [215903] LOCATION:  remote_connections_xact_cleanup, connection.c:2602
```

### How can we reproduce the bug?

```bash
Create a distributed hypertable.
Inserted date into it. Use: COPY public.sec(tagno, timestamp, value, confidence) FROM STDIN BINARY;
```

**Comment by wrona100** [comment]
Updating: the problem only applies to inserting new rows when adding a row involves creating a new chunk. If a chunk exists then adding a row is successful regardless of whether the row is chunk is compressed or not.

**Comment by wrona100** [comment]
I managed to get the issue. Whether this is a bug or a deliberate action, judge for yourself. 
The problem most likely resulted from the fact that I deleted a chunk of the table using the drop_chunk function. In addition, a continuous aggregation was "overlaid" on the table. 
Tables were not cleaned up when deleting:
_timescaledb_catalog.chunk_constraint, 
_timescaledb_catalog.dimension_slice,
_timescaledb_catalog.chunk. 
Only in _timescaledb_catalog.chunk individual chunks are marked as droped.
Such a state of affairs made it impossible to insert new records at the time of deleted chunks.
After deleting the entries, the database for the deleted chunks, the database returned to its functionality, before I did it, to be sure, I removed the continuous aggregations without knowing how it would affect their operation.
Closing, but I feel like removing chunks should be handled better.
